frontend/src/components/account/UserAvatar.tsx

```diff
@@ -0,0 +1,194 @@
+import { Avatar, Theme } from "@mui/material";
+import React, { useContext, useRef, useState } from "react";
+import makeStyles from "@mui/styles/makeStyles";
+import AddAPhotoIcon from "@mui/icons-material/AddAPhoto";
+import CloseIcon from "@mui/icons-material/Close";
+import {
+  getCompressedJPG,
+  getImageDialogHeight,
+  getResizedImage,
+  whitenTransparentPixels,
+} from "../../../public/lib/imageOperations";
+import UploadImageDialog from "../dialogs/UploadImageDialog";
+import UserContext from "../context/UserContext";
+import getTexts from "../../../public/texts/texts";
+import ConfirmDialog from "../dialogs/ConfirmDialog";
+
+export interface AvatarImage {
+  imageUrl?: string;
+  thumbnailImageUrl?: string;
+}
+
+interface UserAvatarProps {
+  mode: "read" | "edit";
+  imageUrl?: string;
+  thumbnailImageUrl?: string;
+  alternativeText?: string;
+  isNarrowScreen: boolean;
+  onAvatarChanged?: (image?: AvatarImage) => void;
+}
+
+const dimensions = 150;
+const ACCEPTED_IMAGE_TYPES = ["image/png", "image/jpeg"];
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  avatarImage: {
+    width: `${dimensions}px`,
+    height: `${dimensions}px`,
+    border: "4px solid white",
+  },
+  imageOverlay: {
+    position: "absolute",
+    opacity: 0.4,
+    backgroundColor: "white",
+    width: `calc(${dimensions}px - 4px)`,
+    height: `calc(${dimensions}px - 4px)`,
+    borderRadius: "100px",
+  },
+  editIconContainer: {
+    display: "flex",
+    flexDirection: "row",
+    position: "absolute",
+    width: `${dimensions}px`,
+    height: `${dimensions}px`,
+    alignItems: "center",
+    justifyContent: "center",
+    cursor: "pointer",
+    columnGap: theme.spacing(1),
+  },
+  editIcon: {
+    fontSize: "40px",
+    cursor: "pointer",
+  },
+}));
+
+export function UserAvatar(props: UserAvatarProps): JSX.Element {
+  const classes = useStyles();
+  const { locale } = useContext(UserContext);
+  const texts = getTexts({ page: "account", locale: locale });
+
+  const inputFileRef = useRef<HTMLInputElement | null>(null);
+  const closeIconRef = useRef(null);
+
+  
+  const [tempImage, setTempImage] = useState<string | undefined>(undefined);
+  const [dialogStates, setDialogStates] = useState<{
+    uploadOpen: boolean;
+    confirmDeleteOpen: boolean;
+  }>({ uploadOpen: false, confirmDeleteOpen: false });
+
+  const [avatarImage, setAvatarImage] = useState<AvatarImage>({
+    imageUrl: props.imageUrl,
+    thumbnailImageUrl: props.thumbnailImageUrl,
+  });
+
+  const onImageChanged = async (avatarEvent) => {
+    const file = avatarEvent.target.files[0];
+    if (file && file.type) {
+      if (!ACCEPTED_IMAGE_TYPES.includes(file.type)) {
+        alert(texts.please_upload_either_a_png_or_a_jpg_file);
+      }
```

Comment:
Is this still necessary with the input only accepting png and jpeg?

Reply:
sounds legit. If the format of the image is somehow "currupted", the backend should still check the incoming image for security reasons